### DISCUSSION TITLE
Disabling zeek/suricata -- unclear on where to edit configuration

### DISCUSSION THREAD

**Discussion by abesinger** [Security-Onion-Solutions/securityonion#6191] [Unsupported Versions] [answered]
It was nice to see the feature to disable Zeek and Suricata (#5566) implemented.

I'm just a little unclear from the notes on where to put the option.  The notes say

> Current implementation requires the following to be added to the Zeek / Suricata pillar (no Installer support at this time)
> `enabled: true | false`

It's not entirely clear where that option goes. From reading the template file, I *think* what I need to do is add the following to `/opt/so/saltstack/local/pillar/minions/<sensor>.sls`, am I correct?

[ UPDATE 2021-11-12: edited the above path to correct the typo pointed out in the answer below]

```
zeek:
  enabled: false
suricata:
  enabled: false
```

**Comment by abesinger** [comment, marked as answer]
Answered my own question empirically.  Yes, the above works. Also works for steno.  I'll see I can't put in a PR to add this to the documentation.

**Comment by dougburks** [comment]
> I think what I need to do is add the following to /opt/so/conf/saltstack/local/pillar/minions/\<sensor\>.sls

Did you mean `/opt/so/saltstack/local/pillar/minions/<sensor>.sls`?

If you want to create a zeek or suricata pillar for a particular sensor, add it to `/opt/so/saltstack/local/pillar/minions/<sensor>.sls`.

If you want to create a zeek or suricata pillar for all sensors, add it to `/opt/so/saltstack/local/pillar/global.sls`.

Reference:
https://docs.securityonion.net/en/2.3/zeek.html#disabling
https://docs.securityonion.net/en/2.3/suricata.html#disabling
https://docs.securityonion.net/en/2.3/salt.html#configuration

UPDATE 2021/11/12 We had basic information at https://docs.securityonion.net/en/2.3/zeek.html#disabling and https://docs.securityonion.net/en/2.3/suricata.html#disabling and I've updated them to look more like https://docs.securityonion.net/en/2.3/stenographer.html#disabling.

**Reply by abesinger** [reply]
> > I think what I need to do is add the following to /opt/so/conf/saltstack/local/pillar/minions/<sensor>.sls
> 
> Did you mean `/opt/so/saltstack/local/pillar/minions/<sensor>.sls`?

You are correct, my mistake, I should have copy/pasted instead of typing from memory.  I think I'll blame it either too much or not enough coffee. :-)

I'm correcting my original post to avoid misleading anyone who might not read all the way through.

> If you want to create a zeek or suricata pillar for a particular sensor, add it to `/opt/so/saltstack/local/pillar/minions/<sensor>.sls`.
> 
> If you want to create a zeek or suricata pillar for all sensors, add it to `/opt/so/saltstack/local/pillar/global.sls`.

Thanks.  I'm still a little fuzzy where/when files get processed by Salt, and which take precedence.  I've worked with several other configuration management systems (cfengine, Puppet, Ansible), but haven't worked with Salt before now.

> UPDATE 2021/11/12 We had basic information at https://docs.securityonion.net/en/2.3/zeek.html#disabling and https://docs.securityonion.net/en/2.3/suricata.html#disabling and I've updated them to look more like https://docs.securityonion.net/en/2.3/stenographer.html#disabling.

Nice, reads very well.
